src/pymor/core/defaults.py

```diff
@@ -382,12 +388,28 @@ def set_defaults(defaults):
         Dictionary of default values. Keys are the full paths of the default
         values (see :func:`defaults`).
     """
+    from pymor.tools import mpi
+    if mpi._event_loop and mpi.rank0:
+        mpi.call(_set_defaults, defaults)
```

Comment:
Won't this cause an infinite recursion? `mpi.call` will also execute `_set_defaults` on rank 0, which in turn calls `set_defaults` again?

Reply:
No. `set_defaults` is only executed on ranks>0 (see `_set_defaults`).

Reply:
Sorry, you're right ..